lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -42,4 +54,115 @@ def self.retrieve_new_ids(last_id)
 
     mappings
   end
+
+  # Batch-creates prospects in Pardot.
+  # Request is only sent to Pardot when the batch is big enough.
+  # @param [Hash] contact
+  # @param [Boolean] eager_submit
+  # @return [Array] two arrays, one for all submitted prospects and one for Pardot errors
+  def self.batch_create_prospects(contact, eager_submit = false)
+    submissions = []
+    errors = []
+
+    prospect = convert_to_prospect_fields contact
+    @@prospect_batch << prospect
+
+    url = build_batch_url BATCH_CREATE_URL, @@prospect_batch
+    if url.length > URL_LENGTH_THRESHOLD || @@prospect_batch.size == MAX_PROSPECT_BATCH_SIZE || eager_submit
```

Comment:
is there a max time we wait before going ahead and sending data? I.e. could we get into a situation where we wait a long time (days?) before updating data because updates are so few? Also is there risk of losing data because we waited to submit and then there was a crash?

Reply:
Good point! There is no max time wait because this method does not promise to eventually send request to Pardot. I'll update method comment to make that clearer. Will also explain the relationship between this method and its sibling, `batch_create_remaining_prospects`. 

If there is a crash in this method, batch data is not lost because we don't reset the batch until after we have sent request to and have received a response from Pardot.

If the crash is in the caller method, batch data will be gone with its PardotV2 instance. (I've converted this method to an instance method and batch data to an instance variable). However, it's the responsibility of the caller to retry or resume its own process.